src/averageAges.js

```diff
@@ -37,7 +44,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const agesOfWomen = people
+    .filter(person => {
+      const isMother = !withChildren || isHasChildren(people, person.name);
+
+      return person.sex === 'f' && isMother;
+    })
+    .map(person => person.died - person.born);
+  const totalAges = agesOfWomen.reduce((a, b) => a + b, 0);
+
+  return (totalAges / agesOfWomen.length) || 0;
```

Comment:
You use `reduce` and division by length in every function. Maybe we should move this logic to a separate helper?